tests/queries/0_stateless/01281_group_by_limit_memory_tracking.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+# Regression for MemoryTracker that had been incorrectly accounted
+# (it was reseted before deallocation)
+#
+# For this will be used:
+# - two-level group by
+# - max_memory_usage_for_user
+# - one users' query in background (to avoid reseting max_memory_usage_for_user)
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+. $CURDIR/../shell_config.sh
+
+set -o pipefail
+
+function execute_null()
+{
+    ${CLICKHOUSE_CLIENT} --format Null -n "$@"
+}
+
+function execute_group_by()
+{
+    local opts=(
+        --max_memory_usage_for_user=$((150<<20))
```

Comment:
Peak memory usage for the main query (with `GROUP BY`) is ~100MiB (with `max_threads=2` as here).
So set `max_memory_usage_for_user` to 150MiB and if the memory tracking accounting will be incorrect then the second query will fail

Note that we also need one running query for the user (`sleep(3)`), since `max_memory_usage_for_user` is installed to 0 once there are no more queries for user.